### ISSUE TITLE
release tarballs don't include submodules

### ISSUE THREAD

**Issue by nzanepro** [materialx/MaterialX#317]
Hi,

If I download a release tar.gz (i.e. [v1.36.4.tar.gz](https://github.com/materialx/MaterialX/archive/v1.36.4.tar.gz)), and run cmake to build and include the option to build the viewer, it fails with this error:

```
CMake Error at source/MaterialXView/CMakeLists.txt:2 (message):
  Building the MaterialX viewer requires the NanoGUI submodule to be present.
  Update your repository by calling the following:

  git submodule update --init --recursive


-- Configuring incomplete, errors occurred!
```

Building from a git clone works great.

Is there a way the submodules could be included with the release tar.gz files?

Thanks,
  Zach

**Comment by jstone-lucasfilm** [comment]
Hi Zach,

If there's a straightforward way to include submodules in GitHub releases, I'd support making this change, but I haven't yet encountered a good solution for this.  Here's a related post on the GitHub community forum, which puts this in the category of feature requests for GitHub:

https://github.community/t5/How-to-use-Git-and-GitHub/Create-Release-that-contains-submodule/td-p/12073

All the best,

-Jonathan

**Comment by nzanepro** [comment]
Hi Jonathan,

So after some digging around and finding a ton of other people wanting the same feature, you are correct, there is no straightforward way to do it.

So I made a way (which hopefully proves good+easy+standalone) to do it here (including the module's source): 
https://github.com/nzanepro/git-archive-submodules
(cd to your repo root, run `git-archive-submodules.sh` from the repo above, voila!)

I've forked this repo and tested adding the resultant tar.gz here: 
https://github.com/nzanepro/MaterialX/releases/tag/v1.36.4-including-submodules

It's simple to add the tar.gz that is output from the git-archive-submodules to the release on the web interface of GitHub. This looks like a great way to automate the upload with the tagged release:
https://github.com/itchio/gothub

This is the piece I needed to get the homebrew formula for MaterialX working, which is now available here: 
https://github.com/nzanepro/homebrew-usd 
(install homebrew on macOS, then `brew install nzanepro/usd/materialx`)

I'm working my way through all the usd dependencies so we can have a full homebrew install of the usd stack. If you decide to incorporate this into your release system, please let me know and I'll delete my fork of MaterialX and repoint homebrew to your release. Maybe I can see if I can work this into GitHub somehow... :-)

Thanks
  Zach

**Comment by jstone-lucasfilm** [comment]
Thanks for this excellent follow-up, Zach, and we'll give your script a try for our next release!

**Comment by jstone-lucasfilm** [comment]
Fixed in MaterialX v1.36.5:
https://github.com/materialx/MaterialX/releases/tag/v1.36.5
